metadeploy/adminapi/api.py

```
@@ -1,10 +1,22 @@
+from ipaddress import IPv4Address
+
 from django.apps import apps
+from django.conf import settings
 from django.utils.decorators import method_decorator
 from django.views.decorators.cache import never_cache
 from rest_framework import pagination, permissions, serializers, viewsets
 from rest_framework.response import Response
 
 
+class IsAllowedIPAddress(permissions.BasePermission):
+    """Permission check for allowed IP networks.
+    """
+
+    def has_permission(self, request, view):
+        ip_addr = IPv4Address(request.META["REMOTE_ADDR"])
+        return any(ip_addr in subnet for subnet in settings.ADMIN_API_ALLOWED_SUBNETS)
```

instead of just returning false which will 403 the view -- actually raise django.core.exceptions.SuspiciousSession, which will 403 the view but also provide a specifically auditable log line